### PR TITLE
feat(build): checkstyle validation when installing

### LINKED ISSUE #133: Add a checkstyle validation on PR
I don't know if it should be part of CI or another tool..

@gastaldi  said:
we can reuse what's used in the wildfly-swarm repositories

https://github.com/wildfly-swarm/checkstyle

The checkstyle config that is used by all projects in the WildFly Swarm organisation
or create our own which may be better I think

### PR DESCRIPTION
**Pull request by ia3andy**
Fixes #133

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -426,6 +426,32 @@
           <serverId>ossrh</serverId>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <version>${version.checkstyle.plugin}</version>
```
Comment (gastaldi) [inline pom.xml]: Where is `version.checkstyle.plugin` declared?

File: pom.xml [hunk 2]
```diff
@@ -426,6 +426,32 @@
...
+        <dependencies>
+          <dependency>
+            <groupId>com.puppycrawl.tools</groupId>
+            <artifactId>checkstyle</artifactId>
+            <version>${version.checkstyle}</version>
```
Comment (gastaldi) [inline pom.xml]: Where is `version.checkstyle` delcared?

File: pom.xml (hunk 1 above)
Comment (ia3andy) [reply pom.xml]: in jboss-parent

Comment (gastaldi) [reply pom.xml]: Ah it's declared in the parent pom. You don't need to declare <version> or the dependency to com.puppycrawl.tools

File: pom.xml (hunk 2 above)
Comment (ia3andy) [reply pom.xml]: it is declared only as a property not as a pluginDependency so it's not working without it.

Comment (ia3andy) [reply pom.xml]: I took this code from there: https://github.com/wildfly-swarm/wildfly-swarm/blob/fb119fbdd3ef22e12d7a3f37df12400407f03d1c/pom.xml#L136

**Comment by gastaldi** [comment]
[test]

**Pull request merged**